### DISCUSSION TITLE
Inheriting @Mappings in different subclasses (but source objects have no common interface/base class)

### DISCUSSION THREAD

**Discussion by rmalaprade** [mapstruct/mapstruct#3553] [Q&A] [answered]
Hi, I have used JAXB2 to generate classes for 5 separate sets of XSDs, which have a lot of common properties/definitions.
So the generated classes all look very similar, but are in different packages (and have different JAXB @XmlType annotations).
_Unfortunately_, JAXB2 does not (or no longer? maybe possible in JAXB1) allow the generated classes to implement a common interface (or extend a common base class).
Also, I have no control over the XSDs.
So I have created 5 `@Mapper` classes, containing a lot of identical `@Mapping`(s). 
Is there any way to define the `@Mapping`(s) in a central class/interface (maybe using generic types) and inherit the `@Mapping`(s) in the concrete `@Mapper` classes? (where I would only map the remaining "different" properties) 

Note that I have read the ["Reusing mapping configurations"](https://mapstruct.org/documentation/stable/reference/html/#_reusing_mapping_configurations) paragraph of the docs, but all the different ways seem to require having a common interface/base class.

**Comment by filiphr** [comment, marked as answer]
I think that what you are looking for is [Mapping Composition](https://mapstruct.org/documentation/stable/reference/html/#mapping-composition).

Yo would be able to define your own annotation and then use that in your mappers.

**Reply by rmalaprade** [reply]
Hi @filiphr, that seems to be *exactly* what I was looking for! thanks!

**Reply by rmalaprade** [reply]
@filiphr quick question: does its "experimental" status means that the feature could possibly disappear in a future version? or just that the "_error messages are not mature yet_", as the doc says ?

**Reply by filiphr** [reply]
It won't disappear. We actually removed the experimental in 1.6.0.Beta1 (#3239)
